api/oc_events.h

```diff
@@ -47,6 +47,10 @@ typedef enum {
   SW_UPDATE_UPGRADING,
   SW_UPDATE_DONE,
 #endif /* OC_SOFTWARE_UPDATE */
+#if defined(OC_PUSH) && defined(OC_SERVER) && defined(OC_CLIENT) &&            \
```

Comment:
This is rather long, what about doing something like I did here -https://github.com/iotivity/iotivity-lite/pull/290/files#diff-99b1c8df5705e6b64ed38894790ccf9a044ff61787ec7cc55a8e4a6d57cf3745 ?

Just copy the file from my PR (I have an issue with CTT test case so I must investigate and cannot merge the PR) and add something like:

```C
#if defined(OC_PUSH) && defined(OC_SERVER) && defined(OC_CLIENT) &&   defined(OC_DYNAMIC_ALLOCATION) && defined(OC_COLLECTIONS_IF_CREATE)
#define OC_HAS_FEATURE_PUSH
#endif
```

And then you can just check for `OC_HAS_FEATURE_PUSH`.

Reply:
good Idea!

Reply:
done